src/snowflake/cli/plugins/nativeapp/lsp_commands.py

```diff
@@ -0,0 +1,88 @@
+from typing import TypedDict
+
+from pygls.server import LanguageServer
+from snowflake.cli.api.output.types import MessageResult
+from snowflake.cli.api.project.definition_manager import DefinitionManager
+from snowflake.cli.plugins.nativeapp.init import nativeapp_init
+from snowflake.cli.plugins.nativeapp.manager import NativeAppManager
+from snowflake.cli.plugins.rpc.manager import LSPPluginContext
+from snowflake.snowpark.dataframe import DataFrame
+
+
+def lsp_plugin(name, version, capabilities):
+    def _decorator(func):
+        func.is_lsp_plugin = True
+        func.lsp_plugin_name = name
+        func.lsp_plugin_version = version
+        func.lsp_plugin_capabilities = capabilities
+        return func
+
+    return _decorator
+
+
+class ConnectionParams(TypedDict):
+    sessionToken: str
+    masterToken: str
+    account: str
+    host: str
+    port: str
+    protocol: str
+
+
+class CreateApplication(TypedDict):
+    workspacePath: str
+    applicationName: str
+    templateRepo: str
+    templateName: str
+
+
+class OpenApplication(TypedDict):
+    project_path: str
+
+
+@lsp_plugin(
+    name="nativeapp",
+    version="0.0.1",
+    capabilities={
+        "createApplication": True,
+        "openApplication": True,
+        "teardownApplication": True,
+    },
+)
+def nade_lsp_plugin(server: LanguageServer, ctx: LSPPluginContext):
+    @server.command("createApplication")
+    def create_app(params: list[CreateApplication]):
+        server.show_message_log(repr(params))
+        project = nativeapp_init(
+            path=params[0]["workspacePath"],
+            name=params[0]["applicationName"],
+            git_url=params[0]["templateRepo"],
+            template=params[0]["templateName"],
```

Comment:
Let's avoid using string keys. Imho dataclasses would be a better choice.

Reply:
I've used dataclasses where I am able to, and kept TypedDicts for the interface where the extension communicates with the language server since they are dicts when they come from the extension.